src/metrics/pocket_area/adaptive_pocket_area.py

```diff
@@ -0,0 +1,108 @@
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+from scipy.spatial import ConvexHull
+
+from src.metrics.pocket_area.base import (
+    InvalidPocketError,
+    PocketArea,
+    PocketAreaMetadata,
+)
+from src.metrics.pocket_area.helpers import (
+    get_distance,
+    get_location,
+    split_records_by_role,
+)
+from src.metrics.pocket_area.passer_radius_area import get_passer_radius_area
+from src.metrics.pocket_area.pocket_pb_ch_area import get_convex_hull
+
+"""
+Adaptive Corvex Hull Pseudocode:
+
+    1. Form a list of the passer, blockers, and rushers
+    2. Find the rusher who is closest to the passer and the distance between them
+    3. Filter out all players whose distance from the passer is greater than that distance
+    4. If the filtered list has more than two players, calculate the convex hull of their coordinates and return the area
+    5. If there are exactly two, calculate the area of a circle with that distance as the radius, then multiply that area by a proportion (let's say 0.25 for now) to represent the cone in front of the passer
+    6. If there are fewer than two players in the filtered list raise an exception because something has gone wrong
+
+"""
+
+
+def find_closest_player_s(
+    point: Dict,
+    blockers: List[Dict],
+    rushers: List[Dict],
+    rusher_difference: float,
+):
+    """Returns the closest players to the given point and the distance
+
+    rusher_difference : acceptable difference parameter from the closest rusher
+    and any other player that should be considered to make the pocket
+    """
+    if not rushers:
+        raise InvalidPocketError("No rusher to guage pocket.")
+
+    closest_players: List[Dict] = []
+    closest_distance = float("inf")
+
+    # Find the closest rusher's distance
+    for player in rushers:
+        d = get_distance(point, player)
+        if d < (closest_distance) or closest_distance == float("inf"):
+            closest_distance = d
+
+    players = blockers + rushers
+
+    # Determine whether each rusher is within the rusher difference error
+    for player in players:
+        d = get_distance(point, player)
+        if d <= (closest_distance + rusher_difference):
+            closest_players.append(player)
+
+    return closest_players, closest_distance
+
+
+# Corvex Hull function with blockers and rushers combined
+def get_convexHull_area(frame: List[Dict]) -> PocketArea:
```

Comment:
Nice helper function. Remember to update the comment to `convex` hull not `corvex` hull.